widget/src/pane_grid/state.rs

```diff
@@ -165,6 +167,53 @@ impl<T> State<T> {
         Some((new_pane, new_split))
     }
 
+    /// Split a target [`Pane`] with a given [`Pane`] on a given [`Region`].
+    ///
+    /// Panes will be swapped by default for [`Region::Center`].
+    pub fn split_with(
+        &mut self,
+        target: &Pane,
+        pane: (T, &Pane),
```

Comment:
Could we take `&Pane` here and then use `HashMap::remove` from `self.panes` to get the state `T` internally?

Reply:
Ah yes! We can get state `T` by closing it first: bc590e2

Much cleaner now and no visual changes to the user